sdcm/provision/azure/kms_provider.py

```diff
@@ -0,0 +1,84 @@
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2025 ScyllaDB
+import logging
+from dataclasses import dataclass
+
+from sdcm.utils.azure_utils import AzureService
+
+LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class KmsProvider:
+    _resource_group_name: str
+    _region: str
+    _az: str
+    _azure_service: AzureService = AzureService()
+    MANAGED_IDENTITY_CONFIG = {
+        'resource_group': 'qa_azure_key_vault',
+        'identity_name': 'myManagedIdentity',
+        'principal_id': '0b5c8c2a-faa1-41d8-ab2d-e9605c2b8654'
+    }
+    SCT_SERVICE_PRINCIPAL_ID = "6c580092-c65c-4527-93d5-25177e10f54e"
```

Comment:
what's that exactly and how it's different from the configuration above ?

Reply:
keyvaults are dynamically created in the SCT temporary resource group where as `qa_azure_key_vault` keeps the static managed identity. SCT principal grants the managed identity access to keyvault

Reply:
Again, please keep this in KeyStore

Reply:
+1 for KeyStore and, also, it looks as `region`-specific thing. Is it?

Reply:
Yes, vault per region. Moved all these static values into keystore and use those values.